src/kdbindings/connection_evaluator.h

```diff
@@ -61,12 +61,30 @@ class ConnectionEvaluator
      */
     void evaluateDeferredConnections()
     {
-        std::lock_guard<std::mutex> lock(m_slotInvocationMutex);
+        std::lock_guard<std::recursive_mutex> lock(m_slotInvocationMutex);
 
-        for (auto &pair : m_deferredSlotInvocations) {
-            pair.second();
+        if (m_isEvaluating) {
+            // We're already evaluating, so we don't want to re-enter this function.
+            return;
         }
+        m_isEvaluating = true;
+
+        // Current best-effort error handling will remove any further invocations that were queued.
```

Comment:
I suggest documenting this in the doc for `evaluateDeferredConnections()`. It's a pretty heavy tool that invalidates calling of unrelated deferred slots so better make the users aware.

Reply:
For now, I've added warnings to the related docs that connection a slot that may throw an exception is undefined behavior.
None of our functions (emit, evaluateDeferred, etc.) deal with this well yet. Even the new error handling is only a band-aid.
So I do not want to document the current behavior, but rather specify that it's undefined.

Reply:
Fair enough